test/common/gitops_utils.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// GitOpsUserSetup configures a new user to use for the GitOps deployments.
+// It returns the path to the new kubeconfig.
+func GitOpsUserSetup(
+	namespace string, username string, additionalRoles ...types.NamespacedName,
+) (OCPUser, string) {
+	const subAdminBinding = "open-cluster-management:subscription-admin"
+
+	ocpUser := OCPUser{
+		ClusterRoles: []types.NamespacedName{
+			{Name: "open-cluster-management:admin:local-cluster"},
+			{
+				Name:      "admin",
+				Namespace: namespace,
+			},
+		},
+		ClusterRoleBindings: []string{subAdminBinding},
+		Password:            "",
+		Username:            username,
+	}
+
+	// Append any additional provided ClusterRoles
+	ocpUser.ClusterRoles = append(ocpUser.ClusterRoles, additionalRoles...)
+
+	// Occasionally, the subscription-admin ClusterRoleBinding may not exist due to some unknown
+	// error. This ClusterRoleBinding is supposed to have been created by the App Lifecycle
+	// controllers. In this unusual case, create the ClusterRoleBinding based on the advice from
+	// the Application Lifecycle squad.
+	subAdminBindingObj := rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: subAdminBinding,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     subAdminBinding,
+		},
+	}
+
+	By("Verifying that the subscription-admin ClusterRoleBinding exists")
+
+	_, err := ClientHub.RbacV1().ClusterRoleBindings().Create(
+		context.TODO(), &subAdminBindingObj, metav1.CreateOptions{},
+	)
+	if err != nil {
+		Expect(k8serrors.IsAlreadyExists(err)).Should(
+			BeTrue(),
+			"Expected error to be 'already exists': "+fmt.Sprint(err),
+		)
+	}
+
+	By("Cleaning up any existing subscription-admin user config")
+	GitOpsCleanup(namespace, "", ocpUser)
+
+	By("Creating a subscription-admin user and configuring IDP")
+	// Create a namespace to house the subscription configuration.
+	nsObj := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_, err = ClientHub.CoreV1().Namespaces().Create(
+		context.TODO(), &nsObj, metav1.CreateOptions{},
+	)
+	Expect(err).Should(BeNil())
+
+	// Create a subscription and local-cluster administrator OpenShift user that can be used
```

Comment:
Since this is a generic helper function.
```suggestion
	// Create the OpenShift user that can be used
```